### DISCUSSION TITLE
how to apply componentwise Dirichlet BC

### DISCUSSION THREAD

**Discussion by RWTHLHK** [gridap/Gridap.jl#1056] [General]
I began using Gridap today, and its functionality has shocked me. I am trying to use Gridap for custom FEM simulation of mechanics problems. One common requirement in simulation is to apply Dirichlet BC componentwise( like y = 0, x, and z are free). I just went through all the tutorials, but I still need help finding a way to do that. Can you explain how to do that for mechanics simulation?

**Comment by JordiManyer** [comment]
Hey @RWTHLHK, welcome to the library.

Have a look at the [elasticity tutorial](https://gridap.github.io/Tutorials/stable/pages/t003_elasticity/#Vector-valued-FE-space-1). The `dirichlet_masks` is what you are looking for: 

```julia
V0 = TestFESpace(model,reffe;
  conformity=:H1,
  dirichlet_tags=["surface_1","surface_2"],
  dirichlet_masks=[(true,false,false), (true,true,true)]
)
```

You setup one triplet per tag, and it will only apply the bcs on the `true` components.

**Reply by RWTHLHK** [reply]
Ah got it！Thanks!
